contracts/src/spaces/facets/entitlements/extensions/EntitlementDataQueryable.sol

```diff
@@ -67,38 +70,56 @@ contract EntitlementDataQueryable is
     bytes32 channelId,
     string calldata permission
   ) internal view returns (Role[] memory) {
+    // retrive the roles associated with the channel
     uint256[] memory channelRoles = ChannelService.getRolesByChannel(channelId);
     uint256 channelRolesLength = channelRoles.length;
 
-    uint256 roleCount = 0;
+    // initialize arrays to store the matching role IDs
     uint256[] memory matchedRoleIds = new uint256[](channelRolesLength);
+    uint256 matchedRoleCount = 0;
 
-    RolesStorage.Layout storage ds = RolesStorage.layout();
+    // access roles storage layout
+    RolesStorage.Layout storage rs = RolesStorage.layout();
 
-    // Count the number of roles that have the requested permission and record their ids.
+    // iterate through channel roles and check for the requested permission
     for (uint256 i; i < channelRolesLength; i++) {
       uint256 roleId = channelRoles[i];
 
-      RolesStorage.Role storage role = ds.roleById[roleId];
+      RolesStorage.Role storage role = rs.roleById[channelRoles[i]];
 
+      // skip immutable roles
```

Comment:
I don't understand why we're skipping these

Reply:
might be legacy code, removing for now.